primitives/trie/src/trie.rs

```diff
@@ -474,6 +474,10 @@ impl<A: Serialize + Deserialize + Clone> MerkleRadixTrie<A> {
     ) -> Vec<TrieNode<A>> {
         let mut chunk = Vec::new();
 
+        if size == 0_usize {
+            return vec![];
```

Comment:
```suggestion
            return chunk;
```
I think that's better since we would re-use the heap allocation. Or maybe we can move the chunk initialization to a line below?